generators/chipyard/src/main/scala/Subsystem.scala

```diff
@@ -87,8 +87,22 @@ trait HasChipyardTilesModuleImp extends HasTilesModuleImp
   val outer: HasChipyardTiles
 }
 
+trait HasHTIF { this: BaseSubsystem =>
+  ResourceBinding {
+    val htif = new Device {
+      def describe(resources: ResourceBindings): Description = {
+        val compat = resources("compat").map(_.value)
+        Description("htif", Map(
+          "compatible" -> compat))
+      }
+    }
+    Resource(htif, "compat").bind(ResourceString("ucb,htif0"))
+  }
+}
+
 class Subsystem(implicit p: Parameters) extends BaseSubsystem
   with HasChipyardTiles
+  with HasHTIF
```

Comment:
Are there any requirements on the configuration for it to support HTIF? Presumably if we built a chipyard config with no front ports of any kind(no tsi, no dmi, no jtag, etc.) htif wouldn't be supported.
I'm not sure its worth figuring this out but it might be worth writing it down somewhere given an external user might stumble across this.

Reply:
We can check for inheritance of the associated traits with a `match` statement.  I'm fine with restricting this to configurations with a known TSI or debug module.  Hypothetically, HTIF could also be implemented without a front bus client by mapping `tohost`/`fromhost` to uncached memory and using a non-coherent DMA engine outside of the `System` level.

Baking HTIF in the device tree (like embedding kernel boot arguments) is itself a bit of a kludge since it is more a property of the simulation/software environment than the hardware.  But without some way to inject a runtime DTB at the bootrom stage, this is the simpler approach.